File: shared-module/src/main/java/dev/atb/models/Credit.java
package dev.atb.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Credit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private float montant;

    @Column(nullable = false)
    private float tauxInteret;

    @Column(nullable = false)
    private int duree;

    private String statut;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "numero_compte_id")
    private Compte numeroCompte;

    @OneToOne(mappedBy = "creditId", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private Credit_model modeldecredit;




}

File: Frontend/src/components/models/Credit.js
class Credit {
    constructor(id, montant, tauxInteret, duree, statut, numeroCompte, modeldecredit) {
        this.id = id;
        this.montant = montant;
        this.tauxInteret = tauxInteret;
        this.duree = duree;
        this.statut = statut;
        this.numeroCompte = numeroCompte;
        this.modeldecredit = modeldecredit;
    }

    static fromDTO(dto) {
        const { id, montant, tauxInteret, duree, statut, numeroCompte, modeldecredit } = dto;
        return new Credit(id, montant, tauxInteret, duree, statut, numeroCompte, CreditModel.fromDTO(modeldecredit));
    }

    toDTO() {
        const { id, montant, tauxInteret, duree, statut, numeroCompte, modeldecredit } = this;
        return { id, montant, tauxInteret, duree, statut, numeroCompte, modeldecredit: modeldecredit.toDTO() };
    }
}

class CreditModel {
    constructor(id, facteursDeRisque, scores, creditId) {
        this.id = id;
        this.facteursDeRisque = facteursDeRisque;
        this.scores = scores;
        this.creditId = creditId;
    }

    static fromDTO(dto) {
        const { id, facteursDeRisque, scores, creditId } = dto;
        return new CreditModel(id, facteursDeRisque, scores, creditId);
    }

    toDTO() {
        const { id, facteursDeRisque, scores, creditId } = this;
        return { id, facteursDeRisque, scores, creditId };
    }
}

export { Credit, CreditModel };
